### PR TITLE
Deprecate in favor of rack-attack

### LINKED ISSUE #43: Deprecate in favor of RackAttack
Hello All!

I wanted to open discussion to see if RackAttack could serve the wonderful people using this lovely gem a bit better. https://github.com/kickstarter/rack-attack

It seems to hame more features/functionality/community than here at rack-throttle & I'm thinking we may want to give up the throne to help peoples looking for this functionality & have come here :)

@bendiken @brandonhilkert + any others, thoughts?

### LINKED ISSUE #55: README in latest release & tagging
**Describe the bug**

This is a minor bug I noticed today, and it likely won't affect many people. When running `solargraph bundle` locally, I got an error while building the yard docs for rack-throttle 0.7.0. 

    No such file or directory @ rb_sysopen - README (Errno::ENOENT)

After doing some digging, I noticed the README in the current published version is a symlink to a file that doesn't exist in the gem (README -> README.md). I couldn't confirm this immediately because of two other minor housekeeping issues:

- Rubygems.org probably needs an update (the gem still points to bendiken/rack-throttle, which correctly forwards to this repo) to reduce confusion
- Published gem versions after 0.5.0 don't appear tagged, or as releases on this repo (this was confusing, coupled with the item above I wasn't sure if there was 2 rack-throttles or not)

**To reproduce**

I was running `solargraph bundle` on a repository that had the `rack-throttle` in the Gemfile, but I think this probably affects anyone trying to builds the docs w/ yard

1. Download the build gem from here: https://rubygems.org/downloads/rack-throttle-0.7.0.gem
2. Extract/open view it however you would normally (I just unzipped everything)
3. Observe a README symlink, but no README.md

image

**Expected behavior**

This appears to be fixed in latest master, but the the latest published version isn't correct

**Additional context**

I don't think there's anything I can contribute, PR wise, since it appears fixed in master but I'm happy to help validate after the new version is published. I can certainly point to master for now but it would be nice to see a new release published & the links updated (and versions tagged in github also)

### PR DESCRIPTION
**Pull request by FreekingDean**
This is a really large conceptual PR so I'm going to let this sit for a while to gather feedback!

RackAttack has much more active maintenance & is more mature. I believe we should nudge those looking to do rate limiting in rack apps that way to be the most helpful we can be!
Fixes #43 
Fixes #55

### CODE REVIEW

**Comment by FreekingDean** [comment]
@gkellogg I noticed a few dependency updates from you & dependabot. I was wondering if you wanted to chime in here?

@artob as well, but I know you have been away from open source stuffs lately :)

**Comment by gkellogg** [comment]
@FreekingDean I don't really have an opinion; dependency updates would be due to the RDF.rb 3.1 release and dropping support for older Ruby versions, and of course versions of Rack < 2.0.

I think the way to probably handle it is to do necessary updates to make this gem current, and add a [DEPRECATION] warning that points people to RackAttack and that this gem is now deprecated and will not receive any substantial updates. [DEPRECATION] probably needs to go in the gemspec post_install_message and all of the `initialize` methods for each module.

As the author and only person who can update RubyGems, @artob would need to either add an owner to the library (preferable) or do the push to RubyGems for whatever comes out of this.

I see it's used by 236 different repos, although the gem spec looks pretty old.

See https://github.com/ruby-rdf/rack-linkeddata/blob/develop/rack-linkeddata.gemspec for something that would be compatible with where we are with Ruby RDF.

**Comment by FreekingDean** [comment]
I have rubygem push access on this! I just didn't want to make a decision
in a vacuum :)
Thanks!

--
Dean Galvin
Software Developer
M: 973-262-2132

File: README.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+DEPRECATED We suggest using rack-attack instead
+===============================================
+
+<https://github.com/kickstarter/rack-attack> Accomplishes the same goal as rack-throttle,
```
Comment (sandstrom) [inline README.md:4]:
```suggestion
<https://github.com/rack/rack-attack> Accomplishes the same goal as rack-throttle,
```

File: lib/rack/throttle/limiter.rb [hunk 2]
```diff
@@ -23,6 +23,7 @@ class Limiter
     # @option options [String]  :message    ("Rate Limit Exceeded")
     # @option options [String]  :type       ("text/plain; charset=utf-8")
     def initialize(app, options = {})
+      warn "[DEPRECATION] `rack-throttle` is deprecated.  Please use consider using `rack-attack` https://github.com/kickstarter/rack-attack instead."
```
Comment (sandstrom) [inline lib/rack/throttle/limiter.rb:26]:
```suggestion
      warn "[DEPRECATION] `rack-throttle` is deprecated.  Please use consider using `rack-attack` https://github.com/rack/rack-attack instead."
```

File: rack-throttle.gemspec [hunk 3]
```diff
@@ -38,5 +38,9 @@ Gem::Specification.new do |gem|
 
   gem.add_runtime_dependency     'rack',      '>= 1.0.0'
 
-  gem.post_install_message       = nil
+  gem.post_install_message       = <<-POST
+rack-throttle is no longer under active development. Please consider
+using https://github.com/kickstarter/rack-attack instead as it is
```
Comment (sandstrom) [inline rack-throttle.gemspec:43]:
```suggestion
using https://github.com/rack/rack-attack instead as it is
```

**Review by sandstrom — COMMENTED** [review]
I think this is a very sound adjustment.

Just one data point, but I found this quickly, but didn't notice rack-attack until quite a while later.

**Comment by FreekingDean** [comment]
Thanks that must be new :)
Thanks!

--
Dean Galvin
Software Developer
M: 973-262-2132

**Comment by athix** [comment]
@dryruby @FreekingDean any movement on this? This would help save some time for developers looking into rate limiting for rack-based applications.

**Pull request merged**